native/membrane_videocompositor_wgpu/src/compositor.rs

```diff
@@ -19,60 +38,20 @@ impl Vertex {
     };
 }
 
-const UPPER_VERTICES: [Vertex; 4] = [
-    Vertex {
-        position: [1.0, 1.0, 0.0],
-        texture_coords: [1.0, 0.0],
-    },
-    Vertex {
-        position: [-1.0, 1.0, 0.0],
-        texture_coords: [0.0, 0.0],
-    },
-    Vertex {
-        position: [-1.0, 0.0, 0.0],
-        texture_coords: [0.0, 1.0],
-    },
-    Vertex {
-        position: [1.0, 0.0, 0.0],
-        texture_coords: [1.0, 1.0],
-    },
-];
-
-const LOWER_VERTICES: [Vertex; 4] = [
-    Vertex {
-        position: [1.0, 0.0, 0.0],
-        texture_coords: [1.0, 0.0],
-    },
-    Vertex {
-        position: [-1.0, 0.0, 0.0],
-        texture_coords: [0.0, 0.0],
-    },
-    Vertex {
-        position: [-1.0, -1.0, 0.0],
-        texture_coords: [0.0, 1.0],
-    },
-    Vertex {
-        position: [1.0, -1.0, 0.0],
-        texture_coords: [1.0, 1.0],
-    },
-];
-
 pub struct State {
     device: wgpu::Device,
-    input_videos: [InputVideo; 2],
+    input_videos: BTreeMap<usize, InputVideo>,
     output_textures: OutputTextures,
     pipeline: wgpu::RenderPipeline,
     queue: wgpu::Queue,
     _sampler: wgpu::Sampler,
     sampler_bind_group: wgpu::BindGroup,
+    texture_bind_group_layout: wgpu::BindGroupLayout,
```

Comment:
Can we keep all information about output textures/sampler in dedicated `sampler` and `output_textures` structs? I think keeping `sampler bind group` and `texture bind group layout` in `State` is not the best solution.

Reply:
I'll do that for sampler but not for textures, since textures change a lot in #43 and this can be managed there.